models/articles.model.js

```diff
@@ -1,5 +1,22 @@
 const db = require("../db/connection");
 
+exports.fetchArticles = () => {
+  return db
+    .query(
+      `SELECT author, title, article_id, topic, created_at,
+      votes, article_img_url, comment_count
+      FROM articles
+      LEFT JOIN
+      (SELECT COUNT(article_id) AS comment_count, article_id AS temp_id
+      FROM comments GROUP BY article_id) temp_table
+      ON articles.article_id = temp_table.temp_id
+      ORDER BY created_at DESC`
+    )
+    .then(({ rows }) => {
+      return rows;
+    });
+};
+
```

Comment:
I'm confused about temp_table - is this a temporary table you have created? For what purpose?

Reply:
I'm joining on a "temp_table" in the sense that it only exists as the output of the SELECT command contained within. The "temp_table" allows me to count all the comments for each article_id, and then join that "temp_table" to the output of everything in "articles", of which I can then select the columns I want. It's how I approached the SQL - admittedly I have about 1 weeks experience so if there is a better / neater way of doing it please let me know!?

Reply:
Ah ok! I think I would just select the columns in the original select along with comments.article_id as a count - it's not how I would approach it, but, you can explain and justify it, so I wouldn't worry about changing it :)

Reply:
Cool, thanks Niamh!